sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -291,11 +355,18 @@ def delete_table(self, **kwargs) -> None:
                 _reprocess_error(decoded_error)
                 raise
 
+    def _encode_key(self, key_name: str, key_value: Any) -> str:
+        try:
+            _, encoded_key = self._encoder(key_name, key_value)
+            return encoded_key
+        except (KeyError, TypeError):
+            return cast(str, key_value)
+
     @overload
     def delete_entity(
         self,
-        partition_key: str,
-        row_key: str,
+        partition_key: Any,
+        row_key: Any,
```

Comment:
do we need to update the docstring for this change?

Reply:
Good catch!